### DISCUSSION TITLE
Is there a command to show all global packages of all the different node versions?

### DISCUSSION THREAD

**Discussion by tallevi008** [coreybutler/nvm-windows#888] [Support]
Hi,
I am using nvm4W v 1.1.10.
Windows 10 Home version 21H1
OS build	19043.2130

I have installed two versions of node * 18.12.1 and 16.18.1
When I was on version 18 I installed nodemon globally but I can only see info regarding nodemon only when I am on that version.
I noticed that it was installed only in the node modules of that version: "C:\Users\Testo\AppData\Roaming\nvm\v18.12.1\node_modules\nodemon"

I wish to be organized and aware for all me global packages.
Is there a command to show me all the global packages that I installed in all the different node version or at least a way to create shared  global npm packages folder for all node versions?

Thanks,
Tal

**Comment by coreybutler** [comment]
No, there is no command for that. I don't think any version managers have a feature like this... yet. This is something that's part of a reporting package planned for [Runtime](https://github.com/coreybutler/nvm-windows/wiki/Runtime), but it won't be ready until Q1 next year. The only way to get that information right now would be to write a custom script to scan your modules.

**Reply by tallevi008** [reply]
Hi Corey,
thank you for the prompt response 😃.
now I saw another issue but I think it could be related.
I noticed that I can only use nodemon only on the node version that installed it with, despite installing it globally according to the nodemon  docs.
Is there a way to make  a folder in which all global npm packages will be installed and could be used in any node version?
(and of course i will know all me global npm packages )

Thank you,
Tal

**Reply by coreybutler** [reply]
Unfortunately, there's no good way to have a universal/shared global modules directory without significant side effects. Some global modules, specifically native ones, depend on the architecture of a specific version of node. Many non-native modules have dependencies on specific versions of Node as well. If, for example, a module updated itself when a new version is used, it may no longer work when you switch back to an older version.

The architecture of npm has also changed significantly over the years. The current approach is to have all modules in a flat structure, but the older approach involved nested modules.

The bottom line is the architecture of npm changes from version to version, so there is no reliable way to share global modules across versions of npm.

The plan to address this in Runtime is with automation tasks built into Runtime, to do things like auto-install global modules on all/some versions of Node, all new versions of node, etc.
